code/controllers/subsystem/ticker.dm

```diff
@@ -422,13 +422,12 @@ var/datum/subsystem/ticker/ticker
 	var/list/total_antagonists = list()
 	//Look into all mobs in world, dead or alive
 	for(var/datum/mind/Mind in minds)
-		var/temprole = Mind.special_role
-		if(temprole)							//if they are an antagonist of some sort.
-			if(temprole in total_antagonists)	//If the role exists already, add the name to it
-				total_antagonists[temprole] += ", [Mind.name]([Mind.key])"
+		for(var/datum/role/R in Mind.antag_roles)							//if they are an antagonist of some sort.
```

Comment:
going to assume this is a list of role datums, in which case, istypeless-forloop